client/src/services/DonuService.ts

```diff
@@ -12,19 +13,85 @@ const callDonu = (request: Donu.Request): Promise<Donu.Response> => {
   }
 };
 
+export const getDonuModelSource = async (model: string, type: Donu.Type): Promise<Donu.ModelGraph> => {
+  const request: Donu.Request = {
+    command: Donu.RequestCommand.GET_MODEL_SOURCE,
+    definition: {
+      type,
+      source: {
+        model,
+      },
+    },
+  };
+  const response = await callDonu(request);
+  if (response.status === Donu.ResponseStatus.success) {
+    const modelSource = response.result.source;
+    return JSON.parse(modelSource);
+  } else {
+    console.error('[DONU Service] — fetchDonuModelSource', response);
+  }
+};
+
+const populateDonuModelWithGrometSource = async (model: Donu.ModelDefinition): Promise<void> => {
+  const gromet = await getDonuModelSource(model.source.model, model.type);
+  (model as any).gromet = gromet;
+};
+
 /** Fetch a complete list of available models from Donu API */
-export const fetchDonuModels = async (): Promise<any[]> => {
+export const fetchDonuModels = async (): Promise<Model.Model[]> => {
   const request: Donu.Request = {
     command: Donu.RequestCommand.LIST_MODELS,
   };
 
   const response = await callDonu(request);
   if (response.status === Donu.ResponseStatus.success) {
-    const models = response?.result as Donu.ModelDefinition[] ?? null;
-    const grometOnlyModels = models.filter(model => {
-      return [Donu.Type.GROMET_PNC/*, Donu.Type.GROMET_FN */].includes(model.type);
+    // 1. Get models
+    let models = response?.result as any ?? null;
+    // 2. Filter models to PNC and FN type
+    models = models.filter(model => {
+      return [Donu.Type.GROMET_PNC, Donu.Type.GROMET_FN].includes(model.type);
     });
-    return donuToModel(grometOnlyModels);
+    // 3. Populate models with model source
+    await Promise.all(models.map(model => populateDonuModelWithGrometSource(model)));
+    // 4. Transform Gromet to graph for rendering
+    models.forEach(async model => {
+      model.graph = GroMEt2Graph.parseGromet(model.gromet);
```

Comment:
Here there will be an exception for emma Gromets. They are too large to just display as they are and we will need to build an aggregate version of them. Nelson has done the aggregation for those gromets so we have two options: 1) put the aggregated version of emmaa gromets in local so we can display them; 2) have a similar parser that we can use on the fly (this will probably be not that viable since Nelson would need to convert that python script to typescript and I think he doesn't know typescript). Thoughts here @adamocarolli @mj3cheun ?

Reply:
We can talk to Nelson about his code and come up with a plan for this. This shouldn't be too much of an issue.

Reply:
Ok, sounds good. I'll start the conversation.